packages/rocketchat-lib/server/lib/sendEmailOnMessage.js

```diff
@@ -114,14 +114,32 @@ RocketChat.callbacks.add('afterSaveMessage', function(message, room) {
 					return;
 				}
 
+				if (RocketChat.settings.get('IMAP_Enable')) {
+					footer = RocketChat.placeholders.replace(RocketChat.settings.get('Email_Footer_Direct_Reply') || '');
+				}
+				const ts = new Date().getTime();
+
 				user.emails.some((email) => {
 					if (email.verified) {
-						email = {
-							to: email.address,
-							from: RocketChat.settings.get('From_Email'),
-							subject: `[${ siteName }] ${ emailSubject }`,
-							html: header + messageHTML + divisorMessage + (linkByUser[user._id] || defaultLink) + footer
-						};
+						if (RocketChat.settings.get('IMAP_Enable')) {
+							email = {
+								to: email.address,
+								from: RocketChat.settings.get('From_Email'),
+								subject: `[${ siteName }] ${ emailSubject }`,
+								headers: {
```

Comment:
Set two headers, 'Reply-To` and `Message-ID` (for backup, might remove after testing).

If your IMAP email is `reply@domain`, `Reply-To` is `reply+messageId@domain`. It would be `From` email while user reply to emails.

`Message-ID` have format `timestamp+roomId+messageId@domain`.